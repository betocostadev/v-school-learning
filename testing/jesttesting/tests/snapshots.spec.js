// SNAPSHOTS
// The Snapshot is a representation of date in string format

const user = {
  name: 'Cornelius Faux',
  age: 69,
  job: 'inventor'
}

// The snapshot of the user would be something like: user = {\"name\": \"Cornelius Faux\", ...}

test('without snapshots', () => {
  // Let's add the string version of the user Object
  const userString = "{\"name\":\"Cornelius Faux\",\"age\":69,\"job\":\"inventor\"}"

  // and compare it accordingly
  expect(JSON.stringify(user)).toBe(userString)
})

test('user matches with snapshot', () => {
  expect(user).toMatchSnapshot()
})