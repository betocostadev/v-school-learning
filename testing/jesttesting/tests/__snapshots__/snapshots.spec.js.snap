// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user matches with snapshot 1`] = `
{
  "age": 69,
  "job": "inventor",
  "name": "Cornelius Faux",
}
`;
